src/modules/ekf2/EKF2.cpp

```diff
@@ -2283,18 +2283,24 @@ void EKF2::UpdateGyroCalibration(const hrt_abstime &timestamp)
 
 void EKF2::UpdateMagCalibration(const hrt_abstime &timestamp)
 {
-	const bool bias_valid = (_ekf.control_status_flags().mag_hdg || _ekf.control_status_flags().mag_3D)
+	const bool bias_valid = (_param_ekf2_mag_type.get() == static_cast<int32_t>(MagFuseType::AUTO)
+				 || _param_ekf2_mag_type.get() == static_cast<int32_t>(MagFuseType::MAG_3D))
+				&& _ekf.control_status_flags().tilt_align
+				&& _ekf.control_status_flags().yaw_align
 				&& _ekf.control_status_flags().mag_aligned_in_flight
+				&& (_ekf.fault_status().value == 0)
 				&& !_ekf.control_status_flags().mag_fault
-				&& !_ekf.control_status_flags().mag_field_disturbed;
+				&& !_ekf.control_status_flags().mag_field_disturbed
+				&& !_ekf.warning_event_flags().stopping_mag_use
+				&& !_ekf.warning_event_flags().emergency_yaw_reset_mag_stopped;
```

Comment:
I don't think we should use the events in there. Also, when the "stopping mag use" event is fired, the `mag_fault` flag is set, so it's redundant.

Reply:
Yes I overdid it, thanks.